src/rust/ensogl/src/math.rs

```diff
@@ -1,5 +1,6 @@
 //! Root module for math-related utilities.
 
+pub mod circle_segment;
```

Comment:
I dont really like the placement of this module. On top of the module we have `topology` and `types`. Having `circle_segment` next to it feels wrong. If this is part of geometry analysis, then it should definitely be somewhere inside of topology, like `topology::geometry::circle::segment` - lets think about it more.

Reply:
I agree that this might be a better fit in a different namespace.
Would topology not sit next to topology, though? I see that a lot of geometric primitives are in the topology module, but my feeling is that topology and geometry are different things. It seems this is also how [Wikipedia structures those topics](https://en.wikipedia.org/wiki/Areas_of_mathematics#Pure_mathematics).

With that in mind, I would propose to put it into `math::geometry::circle::segment`.

Reply:
`Would topology not sit next to topology` - I don't think so :D But regarding geometry, hmm, I need to check it out in more detail.